assets/app/view/game/game_info.rb

```diff
@@ -169,7 +169,7 @@ def trains
         rust_schedule, obsolete_schedule = rust_obsolete_schedule
 
         show_obsolete_schedule = !obsolete_schedule.keys.empty?
-        show_upgrade = @depot.upcoming.any?(&:discount)
+        show_upgrade = @depot.upcoming.map(&:variants).map(&:values).flatten.any? { |train| train['discount'] }
```

Comment:
you don't need the map map flatten, just do this in one any?

any| variants|
   any? variants.values

Reply:
I could swap it for something like this?

```ruby
show_upgrade = @depot.upcoming.any? { |train| train.variants.any? { |_k, v| v['discount'] }}
```

I'm not seeing a way to do it with a single any?  `variants` is always populated, even if there is only a single variant, and sometimes there are multiple variants, but none of them have a discount.